### ISSUE TITLE
AdjustOrderLine to 0 for last orderLine results in error

### ISSUE THREAD

**Issue by martijnvdbrug** [vendure-ecommerce/vendure#1685]
**Describe the bug**
When the activeOrder has only 1 orderline left, and the client changes the amount of the orderline to 0, Vendure returns an error:
`Float cannot represent non numeric value: -Infinity`

**To Reproduce**
1. Add an item to order
2. Execute query (see context for exact query) `adjustOrderLine` with quantity 0
3. See error `Float cannot represent non numeric value: -Infinity`

**Expected behavior**
I would expect the query to return an order without any orderLines

**Environment (please complete the following information):**
 - @vendure/core version: 1.6.4
 - Nodejs version 16
 - Database (mysql/postgres etc): mysql

**Additional context**
Exact error message:
```
{
  "errors": [
    {
      "message": "Float cannot represent non numeric value: -Infinity",
      "locations": [
        {
          "line": 60,
          "column": 13
        }
      ],
      "path": [
        "removeAllOrderLines",
        "taxSummary",
        0,
        "taxRate"
      ]
    }
  ],
  "data": null
}
```
Query used:
```
fragment OrderFields on Order {
  id
  code
  state
  active
  totalWithTax
  subTotalWithTax
  shippingWithTax
  customer {
    id
    firstName
    lastName
    phoneNumber
    emailAddress
  }
  shippingAddress {
    fullName
    company
    streetLine1
    streetLine2
    city
    postalCode
    country
  }
  billingAddress {
    fullName
    company
    streetLine1
    streetLine2
    city
    postalCode
    country
  }
  shippingLines {
    shippingMethod {
      id
      code
      name
    }
  }
  lines {
    id
    quantity
    linePriceWithTax
    featuredAsset {
      id
      preview
      thumbnail
    }
    productVariant {
      id
      sku
      name
      priceWithTax
    }
  }
  taxSummary {
    taxRate
    taxTotal
    taxBase
  }
  payments {
    id
    state
    errorMessage
    metadata
  }
  discounts {
    description
    amountWithTax
  }
  couponCodes
}

mutation adjustOrderLine($orderLineId: ID!, $quantity: Int!) {
  adjustOrderLine(orderLineId: $orderLineId, quantity: $quantity) {
    ... on Order {
      ...OrderFields
    }
    ... on ErrorResult {
      errorCode
      message
    }
  }
}
```

This bug writes the following line to the shippingLines table:
`[{"description":"shipping tax","taxRate":null}]`
This db entry will keep causing issues, even after downgrading to 1.6.2. The error is slightly different after downgrading though:
`Cannot return null for non-nullable field OrderTaxSummary.taxRate`

**Comment by martijnvdbrug** [comment]
`removeOrderLine` results in the same error

**Comment by martijnvdbrug** [comment]
Version `1.6.3` also has this problem

**Comment by michaelbromley** [comment]
Hi Martijn, are you able to reproduce this on a fresh install of 1.6.4?

I'm attempting to reproduce on the `master` branch and cannot.

**Comment by martijnvdbrug** [comment]
The issue is with my custom ShippingCalculator. [This comment](https://github.com/vendure-ecommerce/vendure/issues/1697#issuecomment-1211959203) put me on the right track.

My shipping calculator uses the max taxRate of the order as tax for the shippingmethod:
```
  calculate: (ctx, order, args) => {
    const maxTax = Math.max(...order.lines.map((line) => line.taxRate));
    return {
      price: args.rate,
      taxRate: maxTax,
      priceIncludesTax: true,
    };
  },
```
maxTax resulted in `-Infinity`. Strange thing is that it did work in 1.6.2

TL;DR this issue can be closed
